euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/Context.java

```diff
@@ -16,27 +16,8 @@
 package cz.seznam.euphoria.core.client.io;
 
 /**
- * A collector of elements. Used in functors.
- *
- * @param <T> the type of elements collected through this context
+ * Used in user defined functions to access environment methods.
  */
-public interface Context<T> {
-
-  /**
-   * Collects the given element to the output of this context.
-   *
-   * @param elem the element to collect
-   */
-  void collect(T elem);
-
-  /**
-   * Retrieves the window - if any - underlying the current
-   * execution of this context.
-   *
-   * @return {@code null} if this context is not executed within a
-   *          windowing strategy, otherwise the current window of
-   *          this context
-   */
-  Object getWindow();
+public interface Context extends Environment {
```

Comment:
Ah, I see your point. What I meant is that using overloading for methods of type `UnaryFunctor` and `UnaryFunctionEnv`, because the compiler cannot use type inference there. What is true is that you generally will not mix these two calls in single operator (the operator either uses `UnaryFunction` and `UnaryFunctionEnv` or `UnaryFunctor`. So it's fine. :+1: